Dockerfile

```diff
@@ -0,0 +1,56 @@
+# Based on and inspired by Dockerfiles and resources at:
+#   https://github.com/pytorch/pytorch/blob/master/Dockerfile
+#   https://github.com/anibali/docker-pytorch
+#   https://jtreminio.com/blog/running-docker-containers-as-current-host-user/
+
+ARG BASE_IMAGE=ubuntu:18.04
+ARG PYTHON_VERSION=3.8
+
+FROM ${BASE_IMAGE} AS dev-base
+# Install basic packages.
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    build-essential \
+    curl \
+    git \
+    ca-certificates && \
+    apt-get autoremove -y && \
+    rm -rf /var/lib/apt/lists/*
+ENV PATH /opt/conda/bin:$PATH
+
+FROM dev-base AS conda
+# Install conda.
+ARG PYTHON_VERSION
+ENV CONDA_AUTO_UPDATE_CONDA=false
+RUN curl --silent --show-error --location --output ~/miniconda.sh https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh  && \
+    chmod +x ~/miniconda.sh && \
+    ~/miniconda.sh -b -p /opt/conda && \
+    rm ~/miniconda.sh && \
+    conda install -y python=${PYTHON_VERSION} && \
+    conda clean -ya
+
+FROM conda AS conda-installs
+# Install pytorch for CPU and torchvision.
+ARG PYTORCH_VERSION=1.6.0
+ARG TORCHVISION_VERSION=0.7.0
+ENV NO_CUDA=1
+RUN conda install pytorch==${PYTORCH_VERSION} torchvision==${TORCHVISION_VERSION} cpuonly -y -c pytorch && conda clean -ya
```

Comment:
any specific reason of choosing conda for installation of python and pytorch?

Reply:
I'm pretty much following the [official Pytorch Dockerfile](https://github.com/pytorch/pytorch/blob/master/Dockerfile) here.